### PR TITLE
Move all assets to test-support.js instead of vendor.js

### LINKED ISSUE #397: FastBoot broken without jQuery
After removing jQuery, I get this in a app when it runs in FastBoot:

```
/Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor.js:119501
	var el = document.createElement("fieldset");
	                  ^

TypeError: Cannot read property 'createElement' of undefined
    at assert (/Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor/ecpo-jquery/dist/jquery.js:882:1)
    at /Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor/ecpo-jquery/dist/jquery.js:2701:1
    at /Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor/ecpo-jquery/dist/jquery.js:2752:1
    at /Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor/ecpo-jquery/dist/jquery.js:36:1
    at /Users/simonihmig/Projects/tlw/tmp/broccoli_persistent_filterautoprefixer_filter-output_path-ZtRjUHYd.tmp/assets/vendor/ecpo-jquery/dist/jquery.js:40:1
    at ContextifyScript.Script.runInContext (vm.js:59:29)
    at VMSandbox.eval (/Users/simonihmig/Projects/tlw/node_modules/fastboot/src/vm-sandbox.js:14:16)
    at /Users/simonihmig/Projects/tlw/node_modules/fastboot/src/ember-app.js:156:15
   ...
```

When jQuery is missing, the own version shipping with this addon is imported here: 
https://github.com/san650/ember-cli-page-object/blob/17dbadd4e23310a9948a0767bff6ba395fc9c9ca/index.js#L28-L31

I am not sure why this is breaking FastBoot, while when Ember CLI imports jQuery (when it is not excluded by setting `vendorFiles`), this is not the case!? 🤔

On the other hand, I'm also wondering while jQuery is imported to `vendor.js`, while it is used only for tests. I think it could go into `test-support.js` instead, right?

I quickly tried this out, by adding `, { type: 'test' }` to the `import()` calls. And indeed this fixing the problem (as `test-support.js` is not used in FastBoot), and tests are still running fine. I can easily submit a PR if that seems fine!

The same applies not only to jQuery, but all of this addon's code: couldn't that go into `test-support.js` as well instead of `vendor.js`? There is no point in having the addon's code availabe in dev mode, only in test, or do I miss something. Changing that would be easy as well I think...

### PR DESCRIPTION
**Pull request by simonihmig**
This fixes #397 and also implements the other changes suggested there: All (own and imported) JS is moved from `vendor.js` to `test-support.js`. The repo's code is now in the `addon-test-support` tree, and the trickery in `treeForAddonTestSupport()` (shamelessly stolen from `ember-native-dom-helpers`) makes sure the imports have not changed (i.e. no `test-support/` path added to imports).

I tried this fork with a large app, and was working fine. Didn't have to update anything (except `package.json`), and all still working. I hope I am not missing something though, so please double check if these changes make sense for you! :)

### CODE REVIEW

**Comment by san650** [comment]
Thanks @simonihmig for working on this!

As you said we have to test this change to validate there are not regressions.

I'll test it on some small projects I have but it would be great if @magistrula and others that have big test suites can test it as well.

**Comment by ro0gr** [comment]
@simonihmig I've just [merged](https://github.com/san650/ember-cli-page-object/pull/396) docs updates which produced conflicts in your PR. Sorry for not merging it earlier. Can you please resolve conflicts?

**Comment by ro0gr** [comment]
Hm.. Seems like `addon-test-support/` is not included in the code coverage reports.

File: index.js [hunk 1]
```diff
@@ -21,39 +21,53 @@ module.exports = {
...
+      destDir: `/${this.moduleName()}`,
+      annotation: `Addon#treeForTestSupport (${this.name})`,
+    });
+
+    return this.preprocessJs(namespacedTree, '/', this.name, {
+      registry: this.registry,
+    });
+  },
+
+  _shouldIncludeFiles() {
```
[the remark this answers is not in the document]
Comment (simonihmig) [reply index.js]: Yeah, good catch. Makes sense!

**Comment by simonihmig** [comment]
Thanks for the quick feedback!

I rebased this, and cleaned this up a bit further.
Removed `_shouldIncludeFiles`, and changed the node-assets config a bit, as it does not need to be dynamic anymore. This revealed that `ceibo` was still imported to `vendor.js`, which should be fixed now as well.

> I believe we still have to change API paths for the docs generator as well

I updated this line, but as I am not aware how that works, please double check...

> I'll test it on some small projects I have but it would be great if @magistrula and others that have big test suites can test it as well.

Absolutely! I reran the latest changes on one of our apps, and its 1700+ tests (all included) are still all passing.

> Hm.. Seems like addon-test-support/ is not included in the code coverage reports.

Sorry, I have no idea how that works! 😝 Also wondered how coverage could have changed significantly. Anything I can do on my side to fix this?

**Comment by ro0gr** [comment]
> Also wondered how coverage could have changed significantly. Anything I can do on my side to fix this?

Looks like it should be [fixed](https://github.com/kategengler/ember-cli-code-coverage/pull/160) in the latest of ember-cli-code-coverage. I think we just have to upgrade it.

**Comment by ro0gr** [comment]
> I think we just have to upgrade it.

I've just tried `ember-cli-code-coverage@1.0.0-beta.3` locally. Unfortunatelly it reports zero coverage for each of file in the `addon-test-support/` folder. Probably it's related to the import path tricks borrowed from `ember-native-dom-helpers`.

**Comment by ro0gr** [comment]
@simonihmig FYI I'm currently working on the code coverage issue on top of your branch. The issue is caused by the fact that `ember-cli-code-coverage` expects generated code to have a fixed module path ([`addon-name/test-support`](https://github.com/kategengler/ember-cli-code-coverage/blob/77b5b9201060d634d281f64a5bb7b412773a8b89/index.js#L178)). So the possible solution is to change only entries module names(`index`, `extend`, `macros`) and leave the rest of the files under `ember-cli-page-object/test-support` namespaced.
 
wip branch: https://github.com/ro0gr/ember-cli-page-object/commits/addon-test-support-coverage

**Comment by simonihmig** [comment]
@ro0gr Hm, I see. Thanks for working on this, and for giving feedback! Hope it works out somehow!

**Comment by ro0gr** [comment]
@simonihmig just merged #400. Thanks for the great idea!

**Pull request closed without merging**